From b82270c30158b7a5692304da903f1d01fa67166e Mon Sep 17 00:00:00 2001
From: Louis Marelle <louis.marelle@latmos.ipsl.fr>
Date: Thu, 18 Aug 2022 15:22:09 +0200
Subject: [PATCH 65/70] Scale MOSAIC in-cloud wet scavenging by cldfra_ls

In module_mosaic_wetscav.F, scale the rate of in-cloud wet scavenging by
the large-scale cloud fraction. This reduces in-cloud aerosol scavenging
when the grid cell is partially cloudy, as in the below-cloud
(impaction) scavenging code.
---
 chem/module_mosaic_wetscav.F | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/chem/module_mosaic_wetscav.F b/chem/module_mosaic_wetscav.F
index 47ce280..703797d 100644
--- a/chem/module_mosaic_wetscav.F
+++ b/chem/module_mosaic_wetscav.F
@@ -326,7 +326,7 @@ CONTAINS
 
       do 100 k=kts,kte
       do 100 i=its,ite
-         scale=1.0-dtstepc*qlsink(i,k,j)
+         scale=1.0-dtstepc*qlsink(i,k,j)*cldfra(i,k,j)
          scale=max(0.0,min(1.0,scale)) ! make sure 0 <= scale <= 1
          if (qlsink(i,k,j) >  0.0) then
             pdel_fac = pdel(i,k,j)/g
@@ -356,7 +356,7 @@ CONTAINS
       do 270 j = jts,jte
       do 270 k = kts,kte
       do 270 i = its,ite
-         fracscav = dtstepc*qlsink(i,k,j)*gas_aqfrac(i,k,j,l)
+         fracscav = dtstepc*qlsink(i,k,j)*gas_aqfrac(i,k,j,l)*cldfra(i,k,j)
          if (fracscav > 0.0) then ! make sure fracscav > 0
             fracscav = min(1.0,fracscav) ! make sure fracscav <= 1
             scale = 1.0 - fracscav
-- 
2.31.1

